### DISCUSSION TITLE
Experimental Decorator/s support

### DISCUSSION THREAD

**Discussion by 1a35e1** [biomejs/biome#854] [General]
Hi all, 

Great tool, took it for a spin on a pretty large NestJS project. Given `experimentalDecorators` has been on the standards track for a while I can appreciate, why some this is not working 100% yet.

In NestJS, its relatively common to have parameters injected in the `constructor()` or a `method()`

It would be good if we could have a way to disable this warning/error while the API is in flux. 

Currently I get `Decorators are not valid here.biome(parse)` which would be nice to ignore for now.

<img width="934" alt="Screenshot 2023-11-23 at 15 41 07" src="https://github.com/biomejs/biome/assets/42714653/829b72ff-f28a-4ce0-872e-27959c614b01">

Will be trying this on a ReactNative project next.

Thanks for the great work!

**Comment by ematipico** [comment]
We support experimental decorators: https://biomejs.dev/reference/configuration/#javascriptparserunsafeparameterdecoratorsenabled

:)

**Comment by 1a35e1** [comment]
Ah just found it. TYVM. 

Just needed to add this!

```
...
"javascript": {
    "parser": {
      "unsafeParameterDecoratorsEnabled": true
    }
  },
....
```

**Reply by ebarthur** [reply]
For some reason, this is not working for me. I'm using v1.9.4 in a Nest.js project

**Reply by HiChen404** [reply]
Maybe you need to restart the server. 
> Biome: restart LSP server. 

It works for me. (v1.9.4)

**Reply by rfarris-inrhythm** [reply]
If you are in a monorepo, make sure you have the biome file in the root of the project as well, then restart the server. This was my issue. The plugin should detect other biome files, but i get where this could be an issue if you wanted different biome configs in a monorepo.

**Comment by Lultik** [comment]
@1a35e1
UPD: I have the same error in JB Webstorm:

![image](https://github.com/user-attachments/assets/affb3b3c-107a-4863-8173-55c2b05f4d24)

`unsafeParameterDecoratorsEnabled` is enabled:

![image](https://github.com/user-attachments/assets/cef6c34b-b848-4524-89c6-1e022f51a49a)


Restarting the server does not fix anything:

![image](https://github.com/user-attachments/assets/c088c3d7-6087-4ce4-81dd-cf9f36c5585a)

![image](https://github.com/user-attachments/assets/b9dd42f4-2915-45d1-8b0e-02c3f6c0503d)

Restart LSP server form actions fix this issue for 1 second and then this error reappears.
![image](https://github.com/user-attachments/assets/c19817a3-888b-43ee-aeed-59babe7a37dd)

**Reply by ebarthur** [reply]
I dunno why this is happening but this is affecting our project at work.

**Reply by Lultik** [reply]
For now, I see only one solution: disable Biome plugin and run `biome check --fix` manually. 😢

**Comment by andrewdavid48** [comment]
I don't understand why this is closed sine WebStorm plugin still shows this false error.

**Comment by andrewdavid48** [comment]
To provide more context, the times it fails is when I have 2 or more projects open with different configurations. For example, is I have a NestJS project open and a React (vite) project open, one will fail. If NestJS project works correctly, the React project does not. If the React project works correctly, the decorator error show up.

**Reply by ematipico** [reply]
It seems you have some issues related to your editor and how it's set with Biome. It's not related to decorators.

There's no need to revive old discussions. Instead, seek help on Discord or open a new issue if you feel there's a bug (with a reproduction). Thank you!

**Comment by dushmanta05** [comment]
Can I do the same for TypeScript? Like if it's a TS file, should I use Typescript instead of JavaScript in biome file?

**Comment by MateusJSouza** [comment]
I'm getting the same error. I've tried putting these rules in my biome.json, but I still get the error. Could it be a conflict with the settings I have in biome.json?

These are the settings of my biome.json:

![image](https://github.com/user-attachments/assets/bb338aa1-9ba4-4f68-8130-9780dbe12dd5)

**Reply by ematipico** [reply]
Open an issue with a reproduction, we don't revive closed discussions
